crates/proto/src/rustls/mod.rs

```diff
@@ -43,3 +49,36 @@ pub fn client_config() -> ClientConfig {
 
     builder.with_no_client_auth()
 }
+
+// Adopted from https://docs.rs/rustls/0.23.22/src/rustls/server/handy.rs.html#171
+/// A simple [`ResolvesServerCert`] implementation.
+#[derive(Debug)]
+pub struct CertificateAndKey(Arc<CertifiedKey>);
```

Comment:
Let's see if we can solve that upstream:

- https://github.com/rustls/rustls/pull/2337

Reply:
This was merged and released in 0.23.23.

Reply:
Thanks, I've switched to the new API.